### ISSUE TITLE
7.2.1 uses wp_ prefix explicitly in MigrationHelper.php

### ISSUE THREAD

**Issue by jcatello** [woothemes/woocommerce#36121]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Customer upgraded to 7.2.1

PHP log was full with errors:

```WordPress database error Unknown column 'wp_postmeta.post_id' in 'where clause' for query SELECT meta_id FROM wpct_postmeta,
                                        (SELECT DISTINCT post_id FROM wpct_postmeta
                                        WHERE (meta_key = '_billing_country' OR meta_key='_shipping_country') AND meta_value='NZ')
                                        AS states_in_country
                                WHERE (meta_key='_billing_state' OR meta_key='_shipping_state')
                                AND meta_value='GI'
                                AND wp_postmeta.post_id = states_in_country.post_id
                                LIMIT 100 made by do_action_ref_array('action_scheduler_run_queue'), WP_Hook->do_action, WP_Hook->apply_filters, ActionScheduler_QueueRunner->run, ActionScheduler_QueueRunner->do_batch, ActionScheduler_Abstract_QueueRunner->process_action, ActionScheduler_Action->execute, do_action_ref_array('woocommerce_run_update_callback'), WP_Hook->do_action, WP_Hook->apply_filters, WC_Install::run_update_callback, wc_update_721_adjust_new_zealand_states, Automattic\WooCommerce\Database\Migrations\MigrationHelper::migrate_country_states, Automattic\WooCommerce\Database\Migrations\MigrationHelper::migrate_country_states_for_orders
```

however you will notice the prefix `wp_postmeta` instead of `wpct_postmeta` which is set in wp-config.php

### Expected behavior

run queries using the correct db prefix

### Actual behavior

runs on default wp_ prefix

### Steps to reproduce

upgrade to 7.2.1 

`woocommerce/src/Database/Migrations/MigrationHelper.php:  AND wp_postmeta.post_id = states_in_country.post_id
`

### WordPress Environment

```
### WordPress Environment ###

WordPress address (URL): https://thehawaiivacationguide.com
Site address (URL): https://thehawaiivacationguide.com
WC Version: 7.1.1
REST API Version: ✔ 7.1.1
WC Blocks Version: ✔ 8.7.6
Action Scheduler Version: ✔ 3.5.3
Log Directory Writable: ✔
WP Version: ❌ 6.0.3 - There is a newer version of WordPress available (6.1.1)
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.19.8
PHP Version: 7.4.16
PHP Post Max Size: 512 MB
PHP Time Limit: 0
PHP Max Input Vars: 10000
cURL Version: 7.29.0
NSS/3.53.1

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.37-MariaDB
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.2.1
WC Database Prefix: wpct_
Total Database Size: 2202.53MB
Database Data Size: 1330.70MB
Database Index Size: 871.83MB
wpct_woocommerce_sessions: Data: 1.02MB + Index: 0.02MB + Engine InnoDB
wpct_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_woocommerce_downloadable_product_permissions: Data: 2.52MB + Index: 3.81MB + Engine InnoDB
wpct_woocommerce_order_items: Data: 1.52MB + Index: 0.19MB + Engine InnoDB
wpct_woocommerce_order_itemmeta: Data: 4.52MB + Index: 4.03MB + Engine InnoDB
wpct_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpct_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_actionscheduler_actions: Data: 559.53MB + Index: 306.83MB + Engine InnoDB
wpct_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_actionscheduler_logs: Data: 199.75MB + Index: 163.31MB + Engine InnoDB
wpct_aioseo_cache: Data: 0.50MB + Index: 0.03MB + Engine InnoDB
wpct_aioseo_notifications: Data: 0.03MB + Index: 0.06MB + Engine InnoDB
wpct_aioseo_posts: Data: 4.52MB + Index: 0.16MB + Engine InnoDB
wpct_bv_activities_store: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_bv_fw_requests: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_bv_ip_store: Data: 0.31MB + Index: 0.34MB + Engine InnoDB
wpct_bv_lp_requests: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_comments: Data: 9.52MB + Index: 10.08MB + Engine InnoDB
wpct_et_bloom_stats: Data: 12.52MB + Index: 0.00MB + Engine InnoDB
wpct_facetwp_index: Data: 0.13MB + Index: 0.16MB + Engine InnoDB
wpct_imagify_files: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wpct_imagify_folders: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_kbp_countdown_entry: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_kbp_form_entry: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_kbp_form_entrymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_options: Data: 26.36MB + Index: 0.53MB + Engine InnoDB
wpct_pmxe_exports: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_pmxe_google_cats: Data: 0.39MB + Index: 0.00MB + Engine InnoDB
wpct_pmxe_posts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_pmxe_templates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_postmeta: Data: 81.22MB + Index: 37.63MB + Engine InnoDB
wpct_posts: Data: 136.05MB + Index: 3.67MB + Engine InnoDB
wpct_ppress_forms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_ppress_formsmeta: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wpct_ppress_meta_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_prli_clicks: Data: 48.58MB + Index: 94.59MB + Engine InnoDB
wpct_prli_links: Data: 0.08MB + Index: 0.17MB + Engine InnoDB
wpct_prli_link_metas: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_rank_math_404_logs: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_rank_math_internal_links: Data: 2.25MB + Index: 0.31MB + Engine InnoDB
wpct_rank_math_internal_meta: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wpct_rank_math_redirections: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wpct_rank_math_redirections_cache: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_shortpixel_folders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_shortpixel_meta: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_sib_model_contact: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_termmeta: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
wpct_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_term_relationships: Data: 1.30MB + Index: 0.27MB + Engine InnoDB
wpct_term_taxonomy: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wpct_usermeta: Data: 0.17MB + Index: 0.09MB + Engine InnoDB
wpct_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wpct_wc_admin_notes: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wpct_wc_admin_note_actions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wpct_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wc_customer_lookup: Data: 1.52MB + Index: 0.38MB + Engine InnoDB
wpct_wc_download_log: Data: 4.52MB + Index: 4.03MB + Engine InnoDB
wpct_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_wc_order_product_lookup: Data: 1.52MB + Index: 0.89MB + Engine InnoDB
wpct_wc_order_stats: Data: 1.52MB + Index: 0.64MB + Engine InnoDB
wpct_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wpct_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wfblockediplog: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wpct_wfconfig: Data: 3.30MB + Index: 0.00MB + Engine InnoDB
wpct_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wffilemods: Data: 11.55MB + Index: 0.00MB + Engine InnoDB
wpct_wfhits: Data: 1.02MB + Index: 0.05MB + Engine InnoDB
wpct_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpct_wfknownfilelist: Data: 5.52MB + Index: 0.00MB + Engine InnoDB
wpct_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wflogins: Data: 0.38MB + Index: 0.16MB + Engine InnoDB
wpct_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wpct_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wpct_wfstatus: Data: 0.13MB + Index: 0.09MB + Engine InnoDB
wpct_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_woocommerce_exported_csv_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wpil_broken_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wpil_click_data: Data: 192.75MB + Index: 232.33MB + Engine InnoDB
wpct_wpil_ignore_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wpil_keywords: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wpil_keyword_links: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wpil_keyword_select_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_wpil_report_links: Data: 3.02MB + Index: 1.25MB + Engine InnoDB
wpct_wpil_target_keyword_data: Data: 2.50MB + Index: 2.27MB + Engine InnoDB
wpct_wpil_urls: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wpil_url_links: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wpmailsmtp_debug_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wpct_wpr_rocket_cache: Data: 0.11MB + Index: 0.13MB + Engine InnoDB
wpct_wpr_rucss_used_css: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wpct_wpts_activity_log: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wpct_yoast_indexable: Data: 2.52MB + Index: 0.95MB + Engine InnoDB
wpct_yoast_indexable_hierarchy: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wpct_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wpct_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wpct_yoast_seo_links: Data: 1.52MB + Index: 0.64MB + Engine InnoDB

### Post Type Counts ###

acf-field: 17
acf-field-group: 3
activities: 96
attachment: 3078
custom_css: 4
customize_changeset: 1
et_pb_layout: 58
et_pb_layout_stashed: 8
kadence_element: 42
layout: 2
nav_menu_item: 30
oembed_cache: 220
page: 18
post: 219
pretty-link: 139
product: 5
product_variation: 3
revision: 1571
shop_coupon: 3
shop_order: 10162
shop_order_refund: 23
stay: 22
web-story: 22
where_to_stay: 1
wp_global_styles: 3
wpcode: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (34) ###

AdThrive Ads: by AdThrive – 2.4.0
Advanced Custom Fields: by WP Engine – 6.0.6
Better Search Replace: by Delicious Brains – 1.4.1
WordPress Backup & Security Plugin - BlogVault: by Backup by BlogVault – 4.84
Code Snippets: by Code Snippets Pro – 3.2.2
ConvertKit for WooCommerce: by ConvertKit – 1.5.4
ConvertKit: by ConvertKit – 2.0.5
Custom Post Type UI: by WebDevStudios – 1.13.4
FacetWP: by FacetWP
LLC – 4.1.3

Imagify: by Imagify – Optimize Images & Convert WebP – 2.1
Jetpack: by Automattic – 11.6
Kadence Blocks - PRO Extension: by Kadence WP – 1.7.19
Kadence Blocks – Gutenberg Blocks for Page Builder Features: by Kadence WP – 2.4.22
Kadence Pro - Premium addon for the Kadence Theme: by Kadence WP – 1.0.7
Kadence Starter Templates: by Kadence WP – 1.2.18
Link Whisper: by Link Whisper – 2.1.5
MainWP Child: by MainWP – 4.3.0.1
Pretty Links: by Pretty Links – 3.2.8
Rank Math SEO: by Rank Math – 1.0.103.1
Simple Social Icons: by StudioPress – 3.1.1
Grow Social Pro by Mediavine: by Mediavine – 2.16.2
Temporary Login Without Password: by StoreApps – 1.8.1.1
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 1.22.24

Product Customer List for WooCommerce: by Kokomo – 3.1.2
Web Stories: by Google – 1.28.0
WooCommerce Customer/Order/Coupon Export: by SkyVerge – 5.3.3
WooCommerce Stripe Gateway: by WooCommerce – 7.0.1
WooCommerce Shipping & Tax: by WooCommerce – 2.1.0
WooCommerce: by Automattic – 7.1.1
Wordfence Security: by Wordfence – 7.8.2
WordPress Importer: by wordpressdotorg – 0.8
WP All Export: by Soflyy – 1.3.8
WP Mail SMTP: by WPForms – 3.7.0
WP Rocket: by WP Media – 3.12.3.3

### Inactive Plugins (1) ###

Go Live Update Urls: by OnPoint Plugins – 6.4.1

### Dropin Plugins (1) ###

advanced-cache.php: advanced-cache.php

### Must Use Plugins (3) ###

Clear PHP opcode caches: by Jan Reilink – 1.0
ManageWP - Worker Loader: by GoDaddy –
WP CLI Login Command Server: by Evan Mattson – 1.3

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #22 - /best-hawaii-itineraries/
Cart: #5093 - /cart/
Checkout: #5094 - /checkout/
My account: #5095 - /my-account/
Terms and conditions: #38 - /terms-conditions/

### Theme ###

Name: Custom Theme For The Hawaii Vacation Guide
Version: 1.0
Author URL: https://www.katyboykin.com
Child Theme: ✔
Parent Theme Name: Kadence
Parent Theme Version: 1.1.31
Parent Theme Author URL: https://www.kadencewp.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-12-21 23:39:00 -10:00
Options: ✔
Notes: 118
Onboarding: completed

### WooCommerce Customer/Order/Coupon Export ###

Automated exports: 0
Custom formats: 0
Next export: Not scheduled

### Action Scheduler ###

Complete: 891,938
Oldest: 2022-11-21 06:23:12 -1000
Newest: 2022-12-20 20:09:14 -1000

Failed: 70
Oldest: 2022-09-16 10:17:23 -1000
Newest: 2022-12-20 05:18:13 -1000

Pending: 4
Oldest: -0001-11-29 13:28:34 -1031
Newest: 2022-12-21 20:03:34 -1000


### Status report information ###

Generated at: 2022-12-20 20:18:00 -10:00
`
```

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by pacmanito** [comment]
My log is full of DB errors related to above mentioned issue.

**Comment by chrispiepho** [comment]
I think this issue is mentioned here: [36096](https://github.com/woocommerce/woocommerce/issues/36096) 
and fixed for 7.2.2 here:  [36100](https://github.com/woocommerce/woocommerce/pull/36100)

**Comment by barryhughes** [comment]
Thanks for flagging and our apologies for the disruption. Yes, as @chrispiepho notes this was addressed yesterday and released in 7.2.2:

https://github.com/woocommerce/woocommerce/blob/7.2.2/plugins/woocommerce/src/Database/Migrations/MigrationHelper.php#L208
